### ISSUE TITLE
Support for Android Studio?

### ISSUE THREAD

**Issue by dpnkrg** [Kotlin/kotlin-numpy#11]
Can we use this library in the Android Studio?

As of now, I am facing this error?

`Error: Method name '@' in class 'org.jetbrains.numkt.math.ArithmeticOperatorsKt' cannot be represented in dex format.
`

How ho resolve this?

**Comment by devcrocod** [comment]
No, we don't support Android Studio. This library requires CPython and will not work on android. Do you need a math library for android?

This error occurs because dalvik converts files to the .dex format in which infix functions can give such errors.

**Comment by dpnkrg** [comment]
Any similar library?
Can we port this?

**Comment by MKhalusova** [comment]
Hi @dpnkrg! It'd be very helpful to learn what you're trying to achieve. What do you want to do with the library? Without understanding this, it's hard to suggest another library or give any recommendations.

**Comment by jhannink** [comment]
@MKhalusova @devcrocod I have a very similar problem: I would like to use a numpy-like lib in my kotlin projects on mobile (predominantly android) as well as on desktop. Any thoughts on how to link the numpy c-sources in such a way that we don't have to rewrite all the numpy code in kotlin?

For my concrete example projects, the following would suffice:
- nd-arrays with ndim <=2
- arbitrary transforms along axes
- arbitrary reductions along axes
- computing pairswise distances similar to [scipy cdist](https://docs.scipy.org/doc/scipy/reference/generated/scipy.spatial.distance.cdist.html#scipy.spatial.distance.cdist)

**Comment by devcrocod** [comment]
We are exploring the ways to remove Python dependencies.
Thank you for the information about which API is useful for your tasks.

**Comment by dpnkrg** [comment]
Yes, It will be beneficial for us.

**Comment by balazsbanto** [comment]
Hi there! Any progress on this feature? Thanks!

**Comment by dpnkrg** [comment]
Hi, any update on this feature.

**Comment by devcrocod** [comment]
As I said, this project cannot be used in android. But we have a new project [Multik](https://github.com/Kotlin/multik), JVM implementation can be used in Android. In the near future we will support Native for Android.
